### PR TITLE
Do not install VERSION file.

### LINKED ISSUE #9523: Clash with C++ 20 header file version
I recently encountered a problem when compiling some C++ bindings. The problem is that the include path needed for the OCaml header files points normally to the library directory which contains a file name `VERSION`. Unfortunately  C++ introduced a new header file [version](https://en.cppreference.com/w/cpp/header/version) and with a case sensitive file system this results in the OCaml `VERSION` file being included instead of the C++ one.

### PR DESCRIPTION
**Pull request by bschommer**
This PR fixes #9523 by not installing the Version file.

### CODE REVIEW

**Comment by dra27** [comment]
This file started to be installed with OCaml 4.00 (see https://github.com/ocaml/ocaml/commit/44146a1bfd544103e2e49941907f97bbfd0fe38d), though it's not clear from the logs why @damiendoligez changed it. For 4.00 and 4.01, this was the only file installed with the OCaml version (`caml/version.h` was not installed then). Since 4.02, `OCAML_VERSION_STRING` in `caml/version.h` includes the first line of `VERSION`.

**Comment by dra27** [comment]
@bschommer - while it's draft, it would be better to base this on the 4.11 branch (if you've not done it before, you can edit the branch the PR is based on and then force-push your rebase). It will give a better result for @kit-ty-kate differential test.

**Comment by bschommer** [comment]
> @bschommer - while it's draft, it would be better to base this on the 4.11 branch (if you've not done it before, you can edit the branch the PR is based on and then force-push your rebase). It will give a better result for @kit-ty-kate differential test.

I changed the base to 4.11. Thanks for checking the logs to investigate why it was included.

**Comment by kit-ty-kate** [comment]
Sorry for the time this took to test I had a slight issue I had to fix and #9812 to test first.

So! Good news! `opam-health-check` hasn't detected any packages failing due to a lack of the `VERSION` file :smile: 
I think you can go ahead and make a PR for trunk :tada:

**Comment by bschommer** [comment]
Thanks for checking.

**Comment by bschommer** [comment]
I rebased on top of trunk, switched the base for the pull request to trunk and added some Change entry, but I'm not sure if it is in the right part.

**Comment by kit-ty-kate** [comment]
Could you edit the PR description so that it reflects it not being a draft anymore? Just to not get any maintainers confused by this.

**Comment by bschommer** [comment]
> Could you edit the PR description so that it reflects it not being a draft anymore? Just to not get any maintainers confused by this.

Done.

File: Changes [hunk 1]
```diff
@@ -387,6 +387,9 @@ Working version
 - #9824, #9837: Honour the CFLAGS and CPPFLAGS variables.
   (Sébastien Hinderer, review by David Allsopp)
 
+- #9895, #9523: No longer install VERSION file
```
Comment (dra27) [inline Changes]:
```suggestion
- #9895, #9523: Avoid conflict with C++20 by not installing VERSION to the OCaml
  Standard Library directory.
```
(sorry for such a small tweak, but it seems worth being clear _why_ the change is being made as well as _what_ it is!)

Comment (bschommer) [reply Changes]: No problem, I commited your suggested change.

**Comment by bschommer** [comment]
Is there any thing still missing here, it would be nice if that change landed in 4.12.

**Review by dra27 — APPROVED** [review]
There's a typo in the word "Avoid" though 🙈

@xavierleroy - are you happy with this, too? Beyond the (quite understandable) irritation at the change's being necessary, the testing confirms that it should have no impact.

**Comment by bschommer** [comment]
> There's a typo in the word "Avoid" though 

Should now be fixed and the branch is rebased on the current master, sorry for the noise.

**Comment by xavierleroy** [comment]
No problem, thanks for the patch and the discussion.

**Pull request merged**